cmd/launcher/uninstall_linux.go

```diff
@@ -0,0 +1,85 @@
+//go:build linux
+// +build linux
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/pkg/errors"
+)
+
+func removeLauncher(ctx context.Context, logger log.Logger, identifier string) error {
+	if identifier == "" {
+		identifier = "kolide-k2"
+	}
+
+	isDebianOS, err := isDebian(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	serviceName := fmt.Sprintf("launcher.%s", identifier)
+	packageName := fmt.Sprintf("launcher-%s", identifier)
+
+	// Stop launcher service
+	if _, err := tablehelpers.Exec(ctx, logger, 30, []string{"systemctl"}, []string{"stop", serviceName}); err != nil {
```

Comment:
I would have expected the package manager to do this.

Reply:
Looks like we can combine these to into `systemctl disable --now <serviceName>`

> --now
>            When used with enable, the units will also be started. When
>            used with disable or mask, the units will also be stopped.
>            The start or stop operation is only carried out when the
>            respective enable or disable operation has been successful.

https://man7.org/linux/man-pages/man1/systemctl.1.html